json/src/value.rs

```diff
@@ -468,10 +468,14 @@ struct WriterFormatter<'a, 'b: 'a> {
 
 impl<'a, 'b> io::Write for WriterFormatter<'a, 'b> {
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        match self.inner.write_str(str::from_utf8(buf).unwrap()) {
-            Ok(_) => Ok(buf.len()),
-            Err(_) => Err(io::Error::last_os_error()),
```

Comment:
The only callers are the `fmt::Debug` and `fmt::Display` impls below, which immediately do `.map_err(|_| fmt::Error)` to throw away the error. I mention this in the comment inside `io_error`.

Also I doubt that `fmt::Formatter::write_str` or `str::from_utf8` would set `errno` anyway, right?